### ISSUE TITLE
[BUG] timestamp2long fractional second parsing

### ISSUE THREAD

**Issue by rwlee** [rapidsai/custrings#405]
Error when the timestamp2long conversion fails to parse %f in a timestamp inside the timestamp string, causing values to default to 0 with no error message. This bug is consistently reproduce-able when not padding the sub-second digit to match the granularity. Erroneous behavior is confirmed in the C++ version too.

s granularity --> requires 3 digits
ms granularity --> requires 3 digits
us granularity --> requires 6 digits
ns granularity --> requires 9 digits

timestamp2long should be able to parse into any granularity regardless of the number of digits in the sub-second.

```
>>> nvstrings.to_device(["1945-03-08 09:58:48", "1945-03-08 09:58:48.1", "1945-03-08 09:58:48.0", "1945-03-08 09:58:48.2380000", '1947-10-26 14:01:12.12354346', '1945-10-26 14:01:12.762', '1965-10-26 14:01:12.762000000']).timestamp2int("%Y-%m-%d %H:%M:%S.%f", "ns")
[-783180072000000000, 0, 0, 0, 0, 0, -131968727238000000]
```

**Comment by davidwendt** [comment]
This is working as intended.
Since all the strings are processed in parallel, creating an infrastructure so that each individual string can report an error is too much overhead. Error checking for improper format is too costly so we decided just to report 0 time to let the user know this.

**Comment by rwlee** [comment]
So the 0 value as a default value for parsing failure or other error, but I doesn't really explain the logic behind the fractional second parsing.

Parsing "1970-01-01 00:00:00.1" in millisecond granularity could be interpreted as:
1. A fractional value, 1/1000'th of a second or 1 millisecond after midnight (I believe this is standard for unix timestamps)
2. A decimal value, 1/10'th of a second after midnight or 100 milliseconds

The example given will fail to parse, but because timestamp2long resolves "1970-01-01 00:00:00.100000" to 100 it seems like it's interpreting sub-second values as a decimal value. The format string being customizable allows using `%f` without a preceding `.` implying that it's a fractional value and the parse requires/is only considering 3 digits.

Is NVStrings hardcoding the number of digits of fractional second values? I don't think this ideal, but I also don't know underlying issues of switching how those values are parsed.

**Comment by davidwendt** [comment]
Yes, the %f must be 6 digits as described here: https://github.com/rapidsai/custrings/blob/ce37487e021b6928d0321a10cbef43407f8ad08a/docs/source/datetime.md
and here: https://docs.python.org/3.7/library/datetime.html#strftime-and-strptime-behavior
Any more or less digits will be undefined behavior.

**Comment by davidwendt** [comment]
For parsing, looks like the code is expecting different number of digits for `%f` depending on the units.
- `ms` is 3 digits and `ns` is 9 digits
- anything else it expects 6 digits
This means padding 0s to the right is always required.
For `ms` valid values are for example `001` `012` `123` `230` `300` etc
For `ns` valid values are for example `000000001` `700000000` 

Sorry, that is different from what I told you before. I should've looked at the code.
